src/Workspaces/Core/Portable/EmbeddedLanguages/LanguageServices/AggregateEmbeddedDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices
+{
+    /// <summary>
+    /// An <see cref="IEmbeddedDiagnosticAnalyzer"/> built out of many individual 
+    /// <see cref="IEmbeddedDiagnosticAnalyzer"/>s
+    /// </summary>
```

Comment:
for example, both the json and regex language services have multiple analyzers.  For example, an analyzer to report the parsing diagnostics produced in one of their strings, and the analyzer that detects a case where there is a probable json/regex string, which woudl then offer to add the /*lang...*/ comment.  This type lets those analyzers be independent, while still only exposing a single analyzer to roslyn.